tests/specs/events.js

```diff
@@ -0,0 +1,1003 @@
+
+  store.add([
+    {id: 'one'},
+    {id: 'two'}
+  ])
+
+  .then(function () {
+    return store.removeAll()
+  })
+
+  .then(waitFor(function () {
+    return removeEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(removeEvents.length, 2, 'triggers 2 remove events')
+    t.is(removeEvents[0].object.id, 'one', '1st event passes object')
+    t.is(removeEvents[1].object.id, 'two', '2nd event passes object')
+  })
+})
+
+test('store.on("change") with adding one', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var changeEvents = []
+
+  store.on('change', addEventToArray.bind(null, changeEvents))
+
+  store.add({
+    foo: 'bar'
+  })
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 1))
+
+  .then(function () {
+    t.is(changeEvents.length, 1, 'triggers 1 change event')
+    t.is(changeEvents[0].eventName, 'add', 'passes the event name')
+    t.is(changeEvents[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.on("change") with updating one', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var changeEvents = []
+
+  store.add({
+    id: 'test'
+  })
+
+  .then(function () {
+    store.on('change', addEventToArray.bind(null, changeEvents))
+
+    return store.update({
+      id: 'test',
+      foo: 'bar'
+    })
+  })
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 1))
+
+  .then(function () {
+    t.is(changeEvents.length, 1, 'triggers 1 change event')
+    t.is(changeEvents[0].eventName, 'update', 'passes the event name')
+    t.is(changeEvents[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.on("change") with removing one', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var changeEvents = []
+
+  store.add({
+    id: 'test',
+    foo: 'bar'
+  })
+
+  .then(function () {
+    store.on('change', addEventToArray.bind(null, changeEvents))
+
+    return store.remove('test')
+  })
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 1))
+
+  .then(function () {
+    t.is(changeEvents.length, 1, 'triggers 1 change event')
+    t.is(changeEvents[0].eventName, 'remove', 'passes the event name')
+    t.is(changeEvents[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.on("change") with adding one and updating it afterwards', function (t) {
+  t.plan(5)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var changeEvents = []
+
+  store.on('change', addEventToArray.bind(null, changeEvents))
+
+  store.add({
+    id: 'one',
+    foo: 'bar'
+  })
+
+  .then(function () {
+    return store.update({
+      id: 'one',
+      foo: 'baz'
+    })
+  })
+
+  .then(waitFor(function () {
+    return changeEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(changeEvents.length, 2, 'triggers 2 change events')
+    t.is(changeEvents[0].object.foo, 'bar', '1st event passes object')
```

Comment:
@zoepage, in this particular case the `update` is done after the `add` promise has returned, so the events should be triggered in that order, right?